src/codecs/avif/decoder.rs

```diff
@@ -86,13 +117,255 @@ impl<R: Read> AvifDecoder<R> {
+    plane: &dav1d::Plane,
+    stride: usize,
+    width: usize,
+    height: usize,
+) -> Plane16View {
+    let mut y_plane_stride = stride >> 1;
+
+    let mut bind_y = vec![];
+    let plane_ref = plane.as_ref();
+
+    let mut shape_y_plane = || {
+        y_plane_stride = width;
+        bind_y = reshape_plane(plane_ref, stride, width, height);
+    };
+
+    if stride & 1 == 0 {
+        match bytemuck::try_cast_slice(plane_ref) {
+            Ok(slice) => Plane16View {
+                data: std::borrow::Cow::Borrowed(slice),
+                stride: y_plane_stride,
+            },
+            Err(_) => {
+                shape_y_plane();
+                Plane16View {
+                    data: std::borrow::Cow::Owned(bind_y),
+                    stride: y_plane_stride,
+                }
+            }
+        }
+    } else {
+        shape_y_plane();
+        Plane16View {
+            data: std::borrow::Cow::Owned(bind_y),
+            stride: y_plane_stride,
+        }
+    }
+}
+
+/// This is correct to transmute FFI data for Y plane and Alpha plane
+fn transmute_chroma_plane16(
+    plane: &dav1d::Plane,
+    pixel_layout: PixelLayout,
+    stride: usize,
+    width: usize,
+    height: usize,
+) -> Plane16View {
+    let plane_ref = plane.as_ref();
+    let mut chroma_plane_stride = stride >> 1;
+    let mut bind_chroma = vec![];
+
+    let mut shape_chroma_plane = || {
+        chroma_plane_stride = match pixel_layout {
+            PixelLayout::I400 => unreachable!(),
+            PixelLayout::I420 | PixelLayout::I422 => (width + 1) / 2,
+            PixelLayout::I444 => width,
+        };
+        let u_plane_height = match pixel_layout {
+            PixelLayout::I400 => unreachable!(),
+            PixelLayout::I420 => (height + 1) / 2,
+            PixelLayout::I422 | PixelLayout::I444 => height,
+        };
+        bind_chroma = reshape_plane(plane_ref, stride, chroma_plane_stride, u_plane_height);
+    };
+
+    if stride & 1 == 0 {
+        match bytemuck::try_cast_slice(plane_ref) {
+            Ok(slice) => Plane16View {
+                data: std::borrow::Cow::Borrowed(slice),
+                stride: chroma_plane_stride,
+            },
+            Err(_) => {
+                shape_chroma_plane();
+                Plane16View {
+                    data: std::borrow::Cow::Owned(bind_chroma),
+                    stride: chroma_plane_stride,
+                }
+            }
+        }
+    } else {
+        shape_chroma_plane();
+        Plane16View {
+            data: std::borrow::Cow::Owned(bind_chroma),
+            stride: chroma_plane_stride,
+        }
+    }
+}
+
+/// Getting one of prebuilt matrix of fails
+fn get_matrix(
+    david_matrix: dav1d::pixel::MatrixCoefficients,
+) -> Result<YuvStandardMatrix, ImageError> {
+    match david_matrix {
+        // Identity just a stub here, we'll handle it in different way
+        dav1d::pixel::MatrixCoefficients::Identity => Ok(YuvStandardMatrix::Bt709),
```

Comment:
I thought about that. However, at the moment it will introduce a few branches in the code, and this won't actually make things easier.
I'll try to think what I can do here.

Reply:
I'll move `Identity` from here to YUV 4:4:4 to handle that case this way
```rust
pub(crate) fn yuv444_to_rgba8(
    image: YuvPlanarImage<u8>,
    rgba: &mut [u8],
    range: YuvIntensityRange,
    matrix: YuvStandardMatrix,
) -> Result<(), ImageError> {
    if matrix == YuvStandardMatrix::Identity {
        gbr_to_rgba8(image, rgba)
    } else {
        yuv444_to_rgbx_impl::<u8, 4, 8>(image, rgba, range, matrix)
    }
}
```

This still far away from something good, however will remove some branches in decoder.